src/components/Login-Modal/Login-Modal.js

```diff
@@ -1,27 +1,98 @@
-import React, { useState, useEffect } from "react";
+import React, { useState, useEffect, useRef, useCallback } from "react";
 import ModalWithForm from "../ModalWithForm/ModalWithForm";
+import { useFormValidator } from "../../hooks/useFormValidation";
 
-const Login = ({ isOpen, openModal, onClose, onLogin }) => {
-  const [email, setEmail] = useState("");
-  const [password, setPassword] = useState("");
+const Login = ({ isOpen, openModal, onClose, onLogin, submitError }) => {
+  const [emailError, setEmailError] = useState("");
+  const [emailUnderline, setEmailUnderline] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+  const [passwordUnderline, setPasswordUnderline] = useState("");
+  const [isFormValid, setIsFormValid] = useState(false);
+  const [isButtonDisabled, setButtonDisabled] = useState("");
+  const [formError, setFormError] = useState("");
+
+  const { values, isValid, errors, handleChange, resetForm } = useFormValidator(
+    ["login-email", "login-password"]
+  );
+
+  const { "login-email": email, "login-password": password } = values;
+
+  const initialValues = {
+    "login-email": "",
+    "login-password": "",
+  };
+
+  const initialValuesRef = useRef(initialValues);
```

Comment:
You don't need this ref because you have `initialValues` and you can use them in line 82-83 


```js

    resetForm(
      initialValues,
      initialValues,
      true
    );

```